### PR TITLE
Allow `:json => false` to disable Accept header

### PR DESCRIPTION
**Pull request by paulmakepeace**
For those times you don't want the JSON traffic and parsing.

Also allow overriding the Accept header.

Fixes https://github.com/mrkamel/swift_client/issues/11

### CODE REVIEW

File: test/swift_client_test.rb [hunk 1]
```diff
@@ -341,6 +341,18 @@ def test_get_objects
     assert_equal objects, @swift_client.get_objects("container-1", :limit => 2, :marker => "object-2").parsed_response
   end
 
+  def test_get_objects_no_json
+    objects = [
+     "object-2",
+     "object-3",
+    ]
+    objects_as_string = objects.map { |o| "#{o}\n" }.join
```
Comment (mrkamel) [inline test/swift_client_test.rb:349]:
```suggestion
    response = ["object-2", "object-3"].join("\n")
```

imo, no need for 4 lines with not much semantics.

File: test/swift_client_test.rb [hunk 2]
```diff
@@ -341,6 +341,18 @@ def test_get_objects
...
+
+    stub_request(:get, "https://example.com/v1/AUTH_account/container-1?limit=2&marker=object-2").with(:headers => { "X-Auth-Token" => "Token" }).to_return(:status => 200, :body => objects_as_string, :headers => { "Content-Type" => "text/html" })
```
Comment (mrkamel) [inline test/swift_client_test.rb:351]: maybe use `text/plain` instead of `text/html`

Comment (paulmakepeace) [reply test/swift_client_test.rb:351]: That would make more sense to me too; the server actually replies `"content-type"=>["text/html; charset=UTF-8"]` 🤷‍♂️

If I send `... headers: { "Accept" => "text/plain" }` it _still_ replies with `text/html` (🤷‍♂️🤷‍♂️)

One thing that steered me away from manually sending the headers Hash is without more work `headers: { accept: "text/plain" }` won't override `"Accept"` or `"accept"`...

I guess ultimately the question is, do users want to care about "JSON or not", or "setting Accept headers". For me it's definitely the former especially given the vagaries of server implementation as we've seen, and frankly the somewhat complex behavior of that Accept.

I agree the server technically could change its default response but I'm sure that's an edge case and would likely break a bunch more stuff out there...

File: test/swift_client_test.rb (hunk 1 above)
Comment (paulmakepeace) [reply test/swift_client_test.rb]: The server does emit a trailing newline. But good point, I just unrolled the loop 👍

File: test/swift_client_test.rb [hunk 3]
```diff
@@ -341,6 +341,15 @@ def test_get_objects
     assert_equal objects, @swift_client.get_objects("container-1", :limit => 2, :marker => "object-2").parsed_response
   end
 
+  def test_get_objects_no_json
+    objects = ["object-2", "object-3"]
```
Comment (mrkamel) [inline test/swift_client_test.rb:345]: can be removed

File: test/swift_client_test.rb [hunk 4]
```diff
@@ -341,6 +341,15 @@ def test_get_objects
...
+    response = "object-2\nobject-3\n"
+
+    stub_request(:get, "https://example.com/v1/AUTH_account/container-1?limit=2&marker=object-2").with(:headers => { "X-Auth-Token" => "Token" }).to_return(:status => 200, :body => objects_as_string, :headers => { "Content-Type" => "text/html" })
```
Comment (mrkamel) [inline test/swift_client_test.rb:348]:
```suggestion
    stub_request(:get, "https://example.com/v1/AUTH_account/container-1?limit=2&marker=object-2").with(:headers => { "X-Auth-Token" => "Token" }).to_return(:status => 200, :body => response, :headers => { "Content-Type" => "text/html" })
```

Comment (paulmakepeace) [reply test/swift_client_test.rb]: ✅
